src/Microsoft.Azure.AppConfiguration.AspNetCore/AzureAppConfigurationRefreshExtensions.cs

```diff
@@ -30,7 +30,12 @@ public static IApplicationBuilder UseAzureAppConfiguration(this IApplicationBuil
                 throw new InvalidOperationException("Unable to find the required services. Please add all the required services by calling 'IServiceCollection.AddAzureAppConfiguration' inside the call to 'ConfigureServices(...)' in the application startup code.");
             }
 
-            return builder.UseMiddleware<AzureAppConfigurationRefreshMiddleware>();
+            if (builder.ApplicationServices.GetService(typeof(EmptyRefresherProvider)) == null)
```

Comment:
Is it possible we don't make `EmptyRefresherProvider` a public class?

Reply:
I can look into it but I wasn't sure how. Also side note, should it be `EmptyConfigurationRefresherProvider`?

Reply:
We can make it internal and use `InternalsVisibleTo` attribute (like we expose the internals to test assemblies).
https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.internalsvisibletoattribute?view=net-7.0

Reply:
Although it might be weird to expose all internals just for this one type.

Reply:
We get an instance of `IConfigurationRefresherProvider`. Can we ignore if its `Refreshers` is empty?

Reply:
I think it makes sense since if there are no Refreshers the middleware is basically a no-op anyway.